### PR TITLE
Fix rendering of `ingress_hosts` in Contour component

### LINKED ISSUE #416: Error installing contour component
Getting this error when installing the contour component with the latest master [6db87a2](https://github.com/kinvolk/lokomotive/commit/6db87a2bef11d4a9bc770cde1e44455a0c572ce0)

Contour component configuration:
```
component "contour" {
  # Optional arguments
  ingress_hosts = ["*.liger.lokomotive-k8s.net"]
}
```

Error:
```
~/go/src/github.com/kinvolk/lokomotive/lokoctl component apply contour
Applying component 'contour'...
FATA[0000] rendering manifests failed: rendering chart failed: failed decoding values: error converting YAML to JSON: yaml: did not find expected alphabetic or numeric character  args="[contour]" command="lokoctl component apply"
```

### PR DESCRIPTION
**Pull request by ipochi**
Fixes #416 
Rendering of `ingress_hosts` field in values template results in a space delimited list of strings.
```
ingress_hosts: [ test.local cluster.local]
```
This causes error in rendering of helm chart [envoy service template], and also results in an error when wildcard pattern is used in `ingress_hosts`.

### CODE REVIEW

File: assets/components/contour/templates/02-service-envoy.yaml [hunk 1]
```diff
@@ -13,7 +13,7 @@ metadata:
     # the original remote IP address.
     service.beta.kubernetes.io/aws-load-balancer-backend-protocol: tcp
     {{- if .Values.ingressHosts }}
-    external-dns.alpha.kubernetes.io/hostname: '{{- join "," .Values.ingressHosts }}'
+    external-dns.alpha.kubernetes.io/hostname: "{{- join "," .Values.ingressHosts }}"
```
Comment (invidian) [inline assets/components/contour/templates/02-service-envoy.yaml:16]: Is this change needed?

File: pkg/components/contour/manifest.go [hunk 2]
```diff
@@ -22,7 +22,11 @@ monitoring:
 {{- end }}
 
 {{- if .IngressHosts }}
-ingressHosts: {{ .IngressHosts }}
+ingressHosts: [
```
Comment (invidian) [inline pkg/components/contour/manifest.go:25]: this is YAML format, why not:
```
ingressHosts:
{{- range .IngressHosts}}
- "{{ . }}"
{{- end }}
``` ?

File: pkg/components/contour/component_test.go [hunk 3]
```diff
@@ -52,6 +52,15 @@ component "contour" {
 	testRenderManifest(t, configHCL)
 }
 
+func TestRenderManifestWithIngressHostsWildcard(t *testing.T) {
+	configHCL := `
+component "contour" {
+  ingress_hosts = ["*.domain.com"]
```
Comment (invidian) [inline pkg/components/contour/component_test.go:58]: Actually, what breaks the rendering? the wildcard or multiple entires?

Comment (ipochi) [reply pkg/components/contour/component_test.go:58]: wildcard breaks the rendering.

multiple entries didn't throw any error, but were rendered incorrectly.

**Comment by ipochi** [comment]
> LGTM, though it seems the first commit message needs to be updated.

updated commit message.
